llvm/test/CodeGen/AIE/internalize.ll

```diff
@@ -0,0 +1,126 @@
+; RUN: opt -O1 -S -mtriple=aie2-none-unknown-elf -aie-internalize-symbols < %s | FileCheck --check-prefix=ONLYMAIN %s
+; RUN: opt -O1 -S -mtriple=aie2-none-unknown-elf -aie-internalize-symbols -aie-internalize-skip-functions="skipfunc1,skipfunc2,skipfunc3" < %s | FileCheck --check-prefix=FUNCLIST %s
+
+; ONLYMAIN-NOT: @gvar_unused
+; FUNCLIST-NOT: @gvar_unused
+@gvar_unused = addrspace(1) global i32 undef, align 4
+
+; ONLYMAIN: @gvar_used{{.*}}addrspace(1) global i32 undef, align 4
+; FUNCLIST: @gvar_used{{.*}}addrspace(1) global i32 undef, align 4
+@gvar_used = addrspace(1) global i32 undef, align 4
+
+; ONLYMAIN: @gvar_func_ptr{{.*}}global ptr null
+; FUNCLIST: @gvar_func_ptr{{.*}}global ptr null
+@gvar_func_ptr = global ptr null
+
+; ONLYMAIN-NOT: define{{.*}}void @func_unused
+; FUNCLIST-NOT: define{{.*}}void @func_unused
+define void @func_unused(ptr addrspace(1) %out, i32 %val) {
+entry:
+  store volatile i32 %val, ptr addrspace(1) %out
+  ret void
+}
+
+; ONLYMAIN: define{{.*}}void @func_used_noinline
+; FUNCLIST-NOT: define{{.*}}void @func_used_noinline
+define void @func_used_noinline(ptr addrspace(1) %out, i32 %val) #0 {
+entry:
+  store volatile i32 %val, ptr addrspace(1) %out
+  ret void
+}
+
+; ONLYMAIN-NOT: define{{.*}}void @func_used_alwaysinline
+; FUNCLIST-NOT: define{{.*}}void @func_used_alwaysinline
+define void @func_used_alwaysinline(ptr addrspace(1) %out, i32 %val) #1 {
+entry:
+  store volatile i32 %val, ptr addrspace(1) %out
+  ret void
+}
+
+; ONLYMAIN: declare void @decl_func()
+; FUNCLIST-NOT: declare void @decl_func()
+declare void @decl_func()
+
+; ONLYMAIN: define{{.*}}void @main()
+; ONLYMAIN: call{{.*}}void @func_used_noinline
+; ONLYMAIN: store volatile
+; ONLYMAIN: call{{.*}}void @decl_func()
+; ONLYMAIN: ret void
+; FUNCLIST-NOT: define{{.*}}void @main()
+; FUNCLIST-NOT: call{{.*}}void @func_used_noinline
+; FUNCLIST-NOT: store volatile
+; FUNCLIST-NOT: call{{.*}}void @decl_func()
+; FUNCLIST-NOT: ret void
```

Comment:
Nit: I think just `; FUNCLIST-NOT: define{{.*}}void @main()` is fine

Reply:
done